db/hybrid_logical_vector.go

```diff
@@ -283,3 +304,11 @@ func (hlv *HybridLogicalVector) computeMacroExpansions() []sgbucket.MacroExpansi
 	}
 	return outputSpec
 }
+
+// addToPreviousVersions will take a source/version pair and add it to the HLV previous versions map
+func (hlv *HybridLogicalVector) addToPreviousVersions(source string, version uint64) {
```

Comment:
I'd name this 'setPreviousVersion()' or possibly 'upsertPreviousVersion()' - add usually implies the operation should fail if a value already exists for the key.